docs/src/snap/explanation/ingress.md

```diff
@@ -0,0 +1,79 @@
+# Ingress
+
+In Kubernetes, understanding how inbound traffic is managed inside of your
+cluster can be complex.
+While there is abundant documentation, this explanation provides you with the essentials
+to successfully manage your Canonical Kubernetes cluster.
+
+## Kubernetes Pods and Services
+
+In Kubernetes, the smallest unit is a pod, which encapsulates application containers.
+Since pods are ephemeral and their IP addresses change when destroyed and restarted,
+they are exposed through services.
+Services offer a stable network interface by providing discoverable names and
+load balancing functionality for managing a set of pods.
+For further details on Kubernetes Services,
+refer to the upstream documentation: [Kubernetes Service][Service].
+
+## Ingress
+
+[Ingress][Ingress K8s] is a Kubernetes Resource that manages
+external access by handling both HTTP and HTTPS traffic to services within your cluster.
+Traffic routed through the Ingress is directed to a service,
+which in turn forwards it to the relevant pod
+running the desired application within a container.
+
+The Ingress Resource lets you define rules on how traffic should get handled.
+Learn more about these rules and how to implement them here:
+[Ingress Rules][Ingress Rules].
+
+While the Ingress Resource manages the routing rules for the incoming traffic,
+the [Ingress Controller][Ingress Controller] is responsible for implementing
+those rules by configuring the underlying networking infrastructure of the cluster.
+Ingress does not work without an Ingress Controller.
+
+The Ingress Controller also serves as a layer 7 (HTTP/HTTPS) load balancer
+that routes traffic from outside of your cluster to services inside of your cluster.
+Please do not confuse this with the Kubernetes Service LoadBalancer type.
+
+```{kroki} ../../assets/ingress.puml
+```
+
+With Canonical Kubernetes, enabling default Ingress is easy:
+See [default Ingress][Ingress].
+At the moment of writing, enabling default Ingress will create a
+[Cilium Ingress Controller][Cilium Ingress Controller] for you.
+If your cluster requires different Ingress Controllers,
+the responsibility of implementation falls upon you.
+
```

Comment:
I think it would be useful to add a note here which more specifically calls out that the underlying mechanism  provided by default will always operate in the same way but may not always be Cilium (if indeed, that is accurate!)

Reply:
So this one is a bit of a slippery slope. If we talk about it it may sound like we may change it some time. Have a look at my change and see what you think. @bschimke95 @evilnick you might have a different opinion?

Reply:
"We reserve the right to..." sounds more like we are definitely planning to change it :joy: 

perhaps we should ask @perk

Reply:
I propose the following wording here:
> The underlying mechanism provided by default is currently Cilium. However, it should always be operated through the provided CLI rather than directly. This way, we can provide the best experience for future cluster maintenance and upgrades.

See my suggestion here: https://github.com/canonical/k8s-snap/pull/376/files/76cde6aa0ac7e6d57ead5a8938e0e0fda0deb537..c585437d88e2da9c006e215369d333bb6cf9395f#r1579796674

Reply:
Thank you!